pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,27 +15,243 @@
+
+        n_sites = 4
+
+        mps = ([np.ones((2, 4))] +
+               [np.ones((4, 2, 4)) for _ in range(1, n_sites - 1)] +
+               [np.ones((4, 2))])
+
+        mps_rc = qml.right_canonicalize_mps(mps)
+
+        # Check that the right-canonical definition is fulfilled
+        for i in range(1, n_sites - 1):
+            tensor = mps_rc[i]
+            contraction_matrix = np.tensordot(tensor, tensor.conj(), axes=([1, 2], [1, 2]))
+            assert np.allclose(contraction_matrix, np.eye(tensor.shape[0]))
+
+    .. details::
+        :title: Usage Details
+
+        The input MPS must be a list of :math:`n` tensors :math:`[A^{(0)}, ..., A^{(n-1)}]`
+        with shapes :math:`d_0, ..., d_{n-1}`, respectively. The first and last tensors have rank :math:`2`
+        while the intermediate tensors have rank :math:`3`.
+
+        The first tensor must have the shape :math:`d_0 = (d_{0,0}, d_{0,1})` where :math:`d_{0,0}`
+        and :math:`d_{0,1}`  correspond to the physical dimension of the site and an auxiliary bond
+        dimension connecting it to the next tensor, respectively.
+
+        The last tensor must have the shape :math:`d_{n-1} = (d_{n-1,0}, d_{n-1,1})` where :math:`d_{n-1,0}`
+        and :math:`d_{n-1,1}` represent the auxiliary dimension from the previous site and the physical
+        dimension of the site, respectively.
+
+        The intermediate tensors must have the shape :math:`d_j = (d_{j,0}, d_{j,1}, d_{j,2})`, where:
+
+        - :math:`d_{j,0}` is the bond dimension connecting to the previous tensor
+        - :math:`d_{j,1}` is the physical dimension for the site
+        - :math:`d_{j,2}` is the bond dimension connecting to the next tensor
+
+        Note that the bond dimensions must match between adjacent tensors such that :math:`d_{j-1,2} = d_{j,0}`.
+
+        Additionally, the physical dimension of the site should always be fixed at :math:`2`
+        (since the dimension of a qubit is :math:`2`), while the other dimensions must be powers of two.
+
+        The following example shows a valid MPS input containing four tensors with
+        dimensions :math:`[(2,2), (2,2,4), (4,2,2), (2,2)]` which satisfy the criteria described above.
+
+        .. code-block::
+
+            mps = [
+                np.array([[0.0, 0.107], [0.994, 0.0]]),
+                np.array(
+                    [
+                        [[0.0, 0.0, 0.0, -0.0], [1.0, 0.0, 0.0, -0.0]],
+                        [[0.0, 1.0, 0.0, -0.0], [0.0, 0.0, 0.0, -0.0]],
+                    ]
+                ),
+                np.array(
+                    [
+                        [[-1.0, 0.0], [0.0, 0.0]],
+                        [[0.0, 0.0], [0.0, 1.0]],
+                        [[0.0, -1.0], [0.0, 0.0]],
+                        [[0.0, 0.0], [1.0, 0.0]],
+                    ]
+                ),
+                np.array([[-1.0, -0.0], [-0.0, -1.0]]),
+            ]
+    """
+
+    _validate_mps_shape(mps)
```

Comment:
Just a fly-by comment, Python asserts are disabled when using calling with interpreter's optimization argument `-O`. I wouldn't use `asserts` ever as a load-bearing test for validity for this reason. As long as there's no concerns about this, I'd say it is fine for now, but if validation is required later I'd consider a rewrite and use appropriate exceptions instead for incorrect inputs. Maybe worth adding a backlog item to Shortcut to consider this.

Reply:
Thanks Lee, I'll let the team know 🙌

Reply:
Thanks @KetpuntoG

Reply:
We will add it to the backlog and will try to fix it soon along with other affected templates 👍